File: backend/src/main/java/com/example/informationstand/models/catalog/Place.java
package com.example.informationstand.models.catalog;

import lombok.*;

import jakarta.persistence.*;

@Entity
@Table(name = "places")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class Place {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private String base64Image;

    @Column(nullable = false)
    private double latitude;

    @Column(nullable = false)
    private double longitude;

    @Column(nullable = false)
    private String address;

    private String phoneNumber;

    @Column(nullable = false)
    private String workingHours;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Category category;

    public Place(Category category) {
        this.category = category;
    }
}

File: flutter-app/lib/features/places/models/place.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'place.freezed.dart';
part 'place.g.dart';

@freezed
abstract class Place with _$Place {
  const factory Place({
    int? id,
    required String name,
    required String description,
    required String base64Image,
    required double latitude,
    required double longitude,
    required String address,
    String? phoneNumber,
    required String workingHours,
    required String category,
  }) = _Place;

  factory Place.fromJson(Map<String, dynamic> json) =>
      _$PlaceFromJson(json);
}
